src/ui/public/agg_types/buckets/histogram.js

```diff
@@ -3,13 +3,62 @@ import moment from 'moment';
 import 'ui/validate_date_interval';
 import AggTypesBucketsBucketAggTypeProvider from 'ui/agg_types/buckets/_bucket_agg_type';
 import AggTypesBucketsCreateFilterHistogramProvider from 'ui/agg_types/buckets/create_filter/histogram';
+import FilterBarQueryFilterProvider from 'ui/filter_bar/query_filter';
 import intervalTemplate from 'ui/agg_types/controls/interval.html';
 import minDocCountTemplate from 'ui/agg_types/controls/min_doc_count.html';
 import extendedBoundsTemplate from 'ui/agg_types/controls/extended_bounds.html';
+import autoScaleTemplate from 'ui/agg_types/controls/auto_scale.html';
 export default function HistogramAggDefinition(Private) {
   let BucketAggType = Private(AggTypesBucketsBucketAggTypeProvider);
   let createFilter = Private(AggTypesBucketsCreateFilterHistogramProvider);
+  let queryFilter = Private(FilterBarQueryFilterProvider);
+  const NUM_BUCKETS = 50;
```

Comment:
Maybe this could be exposed as an additional form input when auto_scale is selected?